pallets/pallet-dyn-filter/src/lib.rs

```diff
@@ -0,0 +1,128 @@
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+//! # Dynamic Call Filter
+//!
+//! Enable or disable specific features without a runtime upgrade.
+//!
+//! - [`Pallet`]
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub mod default_weights;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+pub mod setting;
+
+pub use crate::{default_weights::WeightInfo, pallet::*};
+
+#[frame_support::pallet]
+pub mod pallet {
+
+	use frame_support::{
+		pallet_prelude::*,
+		traits::{Contains, StorageVersion},
+	};
+	use frame_system::pallet_prelude::*;
+
+	use crate::{setting::FilterSettings, WeightInfo};
+
+	pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// The origin check for all DID calls inside this pallet.
+		type ApproveOrigin: EnsureOrigin<<Self as frame_system::Config>::Origin>;
+
+		/// TransferCall filters all calls that allow to transfer funds.
+		type TransferCall: Contains<<Self as frame_system::Config>::Call>;
+
+		/// FeatureCall filters all calls that provide the utility of the chain.
+		type FeatureCall: Contains<<Self as frame_system::Config>::Call>;
+
+		/// XcmCall filters all calls that send messages to other chains
+		type XcmCall: Contains<<Self as frame_system::Config>::Call>;
+
+		/// System calls are not filtered. (SystemCall contains all calls that
+		/// are needed for block production, return true if system call)
+		type SystemCall: Contains<<Self as frame_system::Config>::Call>;
+
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::storage]
+	#[pallet::getter(fn filter_setting)]
+	pub type Filter<T> = StorageValue<_, FilterSettings, ValueQuery>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		NewFilterRules { rules: FilterSettings },
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(<T as Config>::WeightInfo::set_filter())]
+		pub fn set_filter(origin: OriginFor<T>, filter: FilterSettings) -> DispatchResult {
+			T::ApproveOrigin::ensure_origin(origin)?;
+
+			Filter::<T>::set(filter);
+			Self::deposit_event(Event::<T>::NewFilterRules { rules: filter });
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Contains<T::Call> for Pallet<T> {
+		/// The provided call goes through if this returns `true`. Else, it
+		/// fails.
+		fn contains(t: &T::Call) -> bool {
+			// System relevant calls cannot be filtered
+			if T::SystemCall::contains(t) {
+				return true;
+			}
+
+			let FilterSettings {
+				transfer_disabled: transfer,
```

Comment:
Can't we simply use `transfer_disabled`? It makes it easier to understand what the combined boolean condition does at the end of this function.

Reply:
Yep wasn't on purpose. Renaming the struct fields didn't rename these bindings here... [f736dde](https://github.com/KILTprotocol/kilt-node/pull/397/commits/f736dde74099f39ad1d2a444cae550a64ffd31f2)